### ISSUE TITLE
update KDE runtime to 5.12

### ISSUE THREAD

**Issue by eszlari** [flathub/org.kde.kontact#19]

**Comment by detrout** [comment]
Is there a variable you can use in flatpack manifests to reference the arch directory?

Once I got 5.12 built, I had trouble with PLAIN authentication again. I resolved it with:

```
--- a/org.kde.kontact.json
+++ b/org.kde.kontact.json
@@ -72,7 +72,7 @@
                 "--enable-otp",
                 "--enable-plain",
                 "--enable-login",
-                "--with-plugindir=/app/lib/sasl2:/usr/lib/sasl2"
+                "--with-plugindir=/app/lib/sasl2:/usr/lib/x86_64-linux-gnu/sasl2"
             ],
             "no-parallel-make": true,
             "sources": [
```

**Comment by detrout** [comment]
Not sure if this is a 5.12 issue or something else but akonadiconsole errors out with 

```"Couldn't register name 'org.kde.akonadiconsole.kdbus-_1_609' with DBUS - another process owns it already!"```

(I'm running it with ```flatpack-builder run --build org.kde.kontact.json bash``` so I can run akonadictl start and adjust various scaling related environment variables before launching kontact.

**Comment by detrout** [comment]
FLATPACK_ARCH didn't work... also somehow in my build there's a sasl2 plugin directory 
```/app/lib/sasl2:/usr/lib/x86_64-linux-gnu/sasl2``` (yes with the colon)

**Comment by danvratil** [comment]
Hi Diane.

yeah, I got stuck on the Kauth install dir. It's a bug in the runtime, but I don't know where to report it and did not have time to look for it :)

Regarding the akonadiconsole, I wonder where the name comes from, because on my desktop it's just org.kde.akonadiconsole - might there be some Flatpak-specific changes to Qt/Frameworks libraries in the runtime? I will probably need to try it myself, but first I need to get the KAuth problem fixed :)

**Comment by detrout** [comment]
Good luck with kalarm/kauth, 

Looks like I got PLAIN sasl2 auth working with the 5.12 runtime using a simple --with-plugindir (And my /app/lib has no extra "/app/lib/sasl2:" directory.

```
-- a/org.kde.kontact.json
+++ b/org.kde.kontact.json
@@ -72,7 +72,7 @@
                 "--enable-otp",
                 "--enable-plain",
                 "--enable-login",
-                "--with-plugindir=/app/lib/sasl2:/usr/lib/sasl2"
+                "--with-plugindir=/app/lib/sasl2"
             ],
             "no-parallel-make": true,
```

I also wanted to try kontact's office365 support, but that looks like it needs qt5networkwauth for an oauth2 implementation. (At least there's a greyed out box labeled "oauth2 office365" in the create custom receive email dialog that suggests I need oauth2 for office365)

Unfortunately it's not obvious how to adjust qmake's installation directories to install into /app. I tried messing with several environment variables, flailing at the .pro files, "qmake -set"ting some variables, using a different qt.conf  and nothing I did would change the targets of make install.

I was able to force make install to work but it looked like the kdepim-runtime didn't find where ever the library and headers ended up (unsurprisingly given the my solution.)

```
@@ -865,6 +865,23 @@
                 }
             ],
             "modules": [
+                {
+                    "name": "libqt5networkauth5",
+                    "buildsystem": "qmake",
                 }
             ],
             "modules": [
+                {
+                    "name": "libqt5networkauth5",
+                    "buildsystem": "qmake",
+                    "build-commands": [
+                        "make",
+                        "find . -name Makefile -exec  sed -i 's/\\$(INSTALL_ROOT)\\/usr\\//\\/app\\//g' {} \\;",
+                        "find . -name '*.cmake' -exec sed -i 's\/\\/usr/\\/app/g' {} \\;",
+                        "find . -name '*.pc' -exec sed -i 's/\\/usr/\\/app/g' {} \\;"
+                    ],
+                    "sources": [
+                        {
+                            "type": "archive",
+                            "url": "https://download.qt.io/official_releases/qt/5.12/5.12.1/submodules/qtnetworkauth-everywhere-src-5.12.1.tar.xz",
+                            "sha256": "090fbb6be35b2f2edf7cc4cb32f0f9c27a9a66defa06f6d23e1397836e31d44c"
+                        }
+                    ]
+                },
                 {
                     "name": "libkolabxml",
                     "config-opts": [
```

(Also thank you to whoever fixed the color scheme so kontact is readable with a dark theme.)

**Comment by detrout** [comment]
Oh I got akonadiconsole to work by adding
```
        "--own-name=org.kde.akonadiconsole.*",
```
to the finish-args

**Comment by danvratil** [comment]
Could you make a PR for the akonadiconsole and the SASL plugindir,  pretty please? :)

**Comment by detrout** [comment]
Sure... Though it might take me until tonight (US/Pacific) to get to it... I have a lot to get done at work today

**Comment by danvratil** [comment]
Fixed in #32
